### PR TITLE
Refactor service account creation tests

### PR DESCRIPTION
**Pull request by rra**
In preparation for refactoring service account creation, switch the
existing service account creation tests to the new syntax.

### CODE REVIEW

File: itests/fe/service_accounts_test.py [hunk 1]
```diff
@@ -27,37 +18,41 @@
     from tests.setup import SetupTest
 
 
-def test_service_account_lifecycle(async_server, browser):  # noqa: F811
-    browser.get(url(async_server, "/groups/user-admins"))
-
-    page = GroupViewPage(browser)
-    page.click_add_service_account_button()
+def test_service_account_lifecycle(tmpdir, setup, browser):
```
Comment (wrodriguez-dbx) [inline itests/fe/service_accounts_test.py]: Not review, just a general question:

What are your thoughts on basing tests on real workflows like this? It's very natural and I think tends to end up reading very nicely, but I've always stayed away from them to prioritize being comprehensive. So here for example, there are a bunch of small corner cases that aren't exercised.

File: tests/ctl/service_account_test.py [hunk 2]
```diff
@@ -1,127 +1,151 @@
...
+    assert service_account is not None
+    assert service_account.user.name == "good-service@a.co"
+    assert service_account.machine_set == "foo +bar -(org)"
+    assert service_account.description == "this is a service account.\n\n it is for testing"
+    group = Group.get(setup.session, name="some-group")
+    assert get_service_accounts(setup.session, group) == [service_account]
+
+    # If the account already exists, creating it again does nothing.
+    #
+    # TODO(rra): Change this behavior to return an error message instead, since it's confusing.
```
Comment (wrodriguez-dbx) [inline tests/ctl/service_account_test.py]: Also general question rather than review:

Do you have a strategy for actually go backing and hitting TODOs? I avoid leaving them in final checked-in code because I'm pretty sure I'll just forget about them, but they do seem very useful in certain cases

File: itests/fe/service_accounts_test.py (hunk 1 above)
Comment (rra) [reply itests/fe/service_accounts_test.py]: I like them. My preferred style is to have one of these tests and then separate tests for the interesting edge cases. The drawback is that the Selenium tests are very slow, so ideally as much of the edge case testing as possible is done at the usecase test level, and then the Selenium test only tests the full workflow happy path and then any UI-specific edge cases.

In general, though, I'm happy to duplicate test coverage to have both edge-case and full workflow tests.

File: tests/ctl/service_account_test.py (hunk 2 above)
Comment (rra) [reply tests/ctl/service_account_test.py]: In general, I agree with you most of the time. In this specific case, subsequent diffs in this same patch series are going to fix the TODOs, so they won't be there long.

I will sometimes leave long-lived TODOs in the code not because I have any specific plans to go back and fix them but because I want to make explicit to someone reading the code that this bit of code is an error, or at least sub-optimal, and just hasn't been fixed. Otherwise, it can be hard to be sure whether the code is actually ideal because of some subtlety that isn't obvious. One can explain that in English, but TODO makes it totally clear that no, this is something that should be fixed, but there just hasn't been time yet.
